litebird_sim/simulations.py

```diff
@@ -1167,6 +1167,38 @@ def describe_mpi_distribution(self) -> Optional[MpiDistributionDescr]:
             mpi_processes=mpi_processes,
         )
 
+    @_profile
+    def nullify_tod(self, component: str = "tod") -> None:
+        """
+        Set the specified component (default: "tod") of all observations to zero.
+
+        This is typically used to zero out Time-Ordered Data (TOD) in-place across
+        all observations.
+
+        Parameters
+        ----------
+        component : str, optional
+            The attribute name of the data to nullify in each observation.
+            Defaults to "tod".
+
+        Raises
+        ------
+        AttributeError
+            If an observation does not have the specified component.
+        """
+        for i, cur_obs in enumerate(self.observations):
+            if not hasattr(cur_obs, component):
+                raise AttributeError(
+                    f"Observation {i} does not have attribute '{component}'"
+                )
+
+            tod = getattr(cur_obs, component)
```

Comment:
This can be rewritten more idiomatically as:

```python
try:
    tod = getattr(cur_obs, component)
except AttributeError:
    raise AttributeError(f"Observation {i} does not have attribute '{component}'")
```

because `getattr` already raises an `AttributeError` exception is `component` does not exist within `cur_obs`.